scaffolding/lib/utils.js

```diff
@@ -0,0 +1,240 @@
+
+function showDescription({ action, controller }) {
+  console.log('API Description:\n');
+
+  if (! fs.existsSync('../documentation/package.json')) {
+    console.error('Unable to show API description because documentation-v1 is not at \'../documentation\'.');
+    return;
+  }
+
+  const actionFile = `../documentation/src/api-documentation/controller-${_.kebabCase(controller)}/${_.kebabCase(action)}.md`;
+  if (! fs.existsSync(actionFile)) {
+    console.error(`Can not find corresponding action file ${actionFile}`);
+    return;
+  }
+
+  const
+    content = fs.readFileSync(actionFile, 'utf8'),
+    regexp = new RegExp(/```javascript\n[\s\S]+```([\s\S]*)/),
+    result = content.match(regexp);
+
+  if ((result || []).length > 0) {
+    console.log(result[1]);
+  } else {
+    console.log('Can not extract action description :(');
+  }
+}
+
+function showSignatures({ language, action, controller }) {
+  console.log('Function signature:\n');
+
+  switch (language) {
+    case 'js':
+      exec(`find node_modules/kuzzle-sdk/src/controllers -name "*.js" | grep ${controller} | xargs cat | grep '${_.camelCase(action)} ('`, display);
+      break;
+
+    case 'cpp':
+      exec(`cat test/bin/sdk-cpp/include/${controller}.hpp | grep ${_.camelCase(action)}`, display);
+      break;
+
+    case 'java':
+      exec(`javap -classpath test/bin/sdk-java/kuzzlesdk-java.jar io.kuzzle.sdk.${_.upperFirst(_.camelCase(controller))} | grep ${_.camelCase(action)}`, displayJava);
+      break;
+
+    case 'go':
+      exec(`cat ~/go/src/github.com/kuzzleio/sdk-go/${controller}/${_.camelCase(action)}.go | grep ${_.upperFirst(_.camelCase(action))}`, display);
+      exec(`cat ~/go/src/github.com/kuzzleio/sdk-go/${controller}/${_.snakeCase(action)}.go | grep ${_.upperFirst(_.camelCase(action))}`, display);
+      break;
+  }
+
+  console.log('\n');
+}
+
+function renderMarkdownTemplate(variables, actionPath) {
+  const
+    actionTemplate = path.join(templatesPath, `action.${variables.language}.md`),
+    destinationFile = path.join(actionPath, 'index.md');
+
+  return renderTemplate(actionTemplate, destinationFile, variables);
+};
+
+function renderSnippetTemplate(variables, actionPath) {
+  const
+    snippetTemplate = path.join(templatesPath, 'usage-snippets',`usage.${variables.language}`),
+    destinationFile = path.join(actionPath, 'snippets', `${_.kebabCase(variables.action)}.${variables.language}`);
+
+  return renderTemplate(snippetTemplate, destinationFile, variables);
+};
+
+function renderSnippetConfigTemplate(variables, actionPath) {
+  const
+    snippetTemplate = path.join(templatesPath, 'usage-snippets','usage.test.yml'),
+    destinationFile = path.join(actionPath, 'snippets', `${_.kebabCase(variables.action)}.test.yml`);
+
+  return renderTemplate(snippetTemplate, destinationFile, variables);
+};
+
+function extractFromFile(file, regexpInfo, regexpInfoFallback) {
+  const content = fs.readFileSync(file, 'utf8')
+  let
+    regexp,
+    result;
```

Comment:
nitpicking: `result` should be defined line 180 instead, to make its scope more local